Source/ACE.Server/Factories/LootGenerationFactory_Jewelry.cs

```diff
@@ -98,79 +99,72 @@ private static WorldObject CreateJewels(int tier, bool isMagical)
         private static WorldObject CreateJewelry(TreasureDeath profile, bool isMagical)
         {
 
-            // 35% chance ring, 35% chance bracelet, 30% chance necklace
-            int ringPercent = 35;
-            int braceletPercent = 35;
+            // 31% chance ring, 31% chance bracelet, 30% chance necklace 8% chance Trinket
+            int ringPercent = 31;
+            int braceletPercent = 31;
             int necklacePercent = 30;
+            int trinketPercent = 8;
 
-            int jewelrySlot = ThreadSafeRandom.Next(0, ringPercent + braceletPercent + necklacePercent);
+            int jewelrySlot = ThreadSafeRandom.Next(0, ringPercent + braceletPercent + necklacePercent + trinketPercent);
             int jewelType;
 
-            switch (jewelrySlot)
-            {
-                case int n when (n <= ringPercent):
-                    jewelType = LootTables.ringItems[ThreadSafeRandom.Next(0, LootTables.ringItems.Length - 1)];
-                    break;
-                case int n when (n <= ringPercent + braceletPercent && n > ringPercent):
-                    jewelType = LootTables.braceletItems[ThreadSafeRandom.Next(0, LootTables.braceletItems.Length - 1)];
-                    break;
-                case int n when (n <= ringPercent + braceletPercent + necklacePercent && n > ringPercent + braceletPercent):
-                    jewelType = LootTables.necklaceItems[ThreadSafeRandom.Next(0, LootTables.necklaceItems.Length - 1)];
-                    break;
-                default:
-                    return null;
-            }
-
-            //int rank = 0;
-            //int skill_level_limit = 0;
+            // Made this easier to read (switch -> if statement)
+            if (jewelrySlot <= 31)
+                jewelType = LootTables.ringItems[ThreadSafeRandom.Next(0, LootTables.ringItems.Length - 1)];
+            else if (jewelrySlot <= 62)
+                jewelType = LootTables.braceletItems[ThreadSafeRandom.Next(0, LootTables.braceletItems.Length - 1)];
+            else if (jewelrySlot <= 92)
+                jewelType = LootTables.necklaceItems[ThreadSafeRandom.Next(0, LootTables.necklaceItems.Length - 1)];
+            else
+                jewelType = LootTables.trinketItems[ThreadSafeRandom.Next(0, LootTables.trinketItems.Length - 1)];
 
             WorldObject wo = WorldObjectFactory.CreateNewWorldObject((uint)jewelType);
 
             if (wo == null)
                 return null;
 
-            wo.SetProperty(PropertyInt.AppraisalLongDescDecoration, 1);
-            wo.SetProperty(PropertyString.LongDesc, wo.GetProperty(PropertyString.Name));
+            wo.AppraisalLongDescDecoration = 1;
+            wo.LongDesc = wo.GetProperty(PropertyString.Name);
```

Comment:
could be shortened to wo.LongDesc = wo.Name

Reply:
opps missed that one